In [1]:
import pandas as pd
from scipy.stats import ttest_ind, chi2_contingency, f_oneway
import matplotlib.pyplot as plt
import seaborn as sns
# Ignore warnings
import warnings
warnings.filterwarnings('ignore')

In [2]:
data=pd.read_csv(r"Football_Data_Merged_Cleaned.csv")
data

appearance_id    game_id  player_id               date_x  \
0       2222536_1321  2222536.0       1321  2012-08-26 00:00:00   
1       2231391_1321  2231391.0       1321  2012-09-01 00:00:00   
2       2231391_1321  2231391.0       1321  2012-09-01 00:00:00   
3       2231397_1321  2231397.0       1321  2012-09-28 00:00:00   
4       2231397_1321  2231397.0       1321  2012-09-28 00:00:00   
...              ...        ...        ...                  ...   
4649  2224542_119169  2757218.0     795738  2016-07-20 12:00:00   
4650  2224542_119169  2757218.0     796429  2016-07-20 12:00:00   
4651  2224542_119169  2757218.0     860812  2016-07-20 12:00:00   
4652  2224542_119169  2757218.0     971613  2016-07-20 12:00:00   
4653  2224542_119169  2757218.0    1163778  2016-07-20 12:00:00   

       player_name_x competition_id_x  yellow_cards  red_cards  goals  \
0     Jermaine Jones               L1           1.0        0.0    0.0   
1     Jermaine Jones               L1           0.0        0.0    1.0   
2     Jermaine Jones               L1           0.0        0.0    1.0   
3     Jermaine Jones               L1           1.0        0.0    0.0   
4     Jermaine Jones               L1           1.0        0.0    0.0   
...              ...              ...           ...        ...    ...   
4649  Fabian Johnson               L1           0.0        0.0    0.0   
4650  Fabian Johnson               L1           0.0        0.0    0.0   
4651  Fabian Johnson               L1           0.0        0.0    0.0   
4652  Fabian Johnson               L1           0.0        0.0    0.0   
4653  Fabian Johnson               L1           0.0        0.0    0.0   

      assists  ...  country_of_birth date_of_birth        sub_position  \
0         1.0  ...           Germany    1981-11-03  Defensive Midfield   
1         0.0  ...           Germany    1981-11-03  Defensive Midfield   
2         0.0  ...           Germany    1981-11-03  Defensive Midfield   
3         0.0  ...           Germany    1981-11-03  Defensive Midfield   
4         0.0  ...           Germany    1981-11-03  Defensive Midfield   
...       ...  ...               ...           ...                 ...   
4649      0.0  ...     United States    2003-08-26  Attacking Midfield   
4650      0.0  ...     United States    2003-11-18          Right-Back   
4651      0.0  ...     United States    2002-04-22         Left Winger   
4652      0.0  ...     United States    2000-06-21  Defensive Midfield   
4653      0.0  ...     United States    2002-06-13      Centre-Forward   

      position_y   foot height_in_cm  market_value_in_eur  \
0       Midfield   both        184.0            1000000.0   
1       Midfield   both        184.0            1000000.0   
2       Midfield   both        184.0            1000000.0   
3       Midfield   both        184.0            1000000.0   
4       Midfield   both        184.0            1000000.0   
...          ...    ...          ...                  ...   
4649    Midfield  right        175.0            4500000.0   
4650    Defender  right        185.0            1500000.0   
4651      Attack  right        182.0             350000.0   
4652    Midfield  right        193.0             150000.0   
4653      Attack  right        193.0            1000000.0   

      highest_market_value_in_eur contract_expiration_date   agent_name  
0                       8000000.0               2024-12-31    Wasserman  
1                       8000000.0               2024-12-31    Wasserman  
2                       8000000.0               2024-12-31    Wasserman  
3                       8000000.0               2024-12-31    Wasserman  
4                       8000000.0               2024-12-31    Wasserman  
...                           ...                      ...          ...  
4649                    4500000.0               2027-06-30    Wasserman  
4650                    3000000.0               2025-12-31  CAA Stellar  
4651                     350000.0    

#  Hypothesis Testing: Event Analysis

# 1. Impact of Attendance on Goals Scored

In [3]:
import pandas as pd
from scipy.stats import ttest_ind

# Define a threshold for high and low attendance
attendance_threshold = data['attendance'].median()  # Using median as threshold

# Create two groups based on the threshold
high_attendance = data[data['attendance'] >= attendance_threshold]['goals']
low_attendance = data[data['attendance'] < attendance_threshold]['goals']

# Perform independent t-test
t_stat, p_value = ttest_ind(high_attendance, low_attendance, equal_var=False)  # unequal variance assumption

# Output the results
print(f"T-Test Results for Goal Scoring Impact:")
print(f"T-Statistic: {t_stat}")
print(f"P-Value: {p_value}")

# Interpretation based on p-value
alpha = 0.05  # significance level
if p_value < alpha:
    print("Reject the null hypothesis: There is a significant difference in the number of goals scored in high attendance vs low attendance matches.")
else:
    print("Fail to reject the null hypothesis: There is no significant difference in the number of goals scored in high attendance vs low attendance matches.")

T-Test Results for Goal Scoring Impact:
T-Statistic: -8.809740362889766
P-Value: 2.3256260388351792e-18
Reject the null hypothesis: There is a significant difference in the number of goals scored in high attendance vs low attendance matches.


In [ ]:
INTERPRETATION : The T-test results show a significant difference in goal scoring between high and low attendance matches. With a T-statistic of -8.80 and a very small P-value (2.33e-18), we reject the null hypothesis. This indicates that attendance likely impacts the number of goals scored.


# Home Advantage question:
Null Hypothesis (H0): Home teams do not score significantly more goals than away teams.
Alternative Hypothesis (𝐻𝑎): Home teams score significantly more goals than away teams.¶

In [5]:
import pandas as pd
from scipy.stats import ttest_ind

# Assuming 'df' is the dataframe containing match data with 'home_club_goals' and 'away_club_goals' columns

# Group data by home and away goals
home_goals = data['home_club_goals']
away_goals = data['away_club_goals']

# Perform independent t-test
t_stat, p_value = ttest_ind(home_goals, away_goals, equal_var=False)  # unequal variance assumption

# Output the results
print(f"T-Test Results for Home Advantage:")
print(f"T-Statistic: {t_stat}")
print(f"P-Value: {p_value}")

# Interpretation based on p-value
alpha = 0.05  # significance level
if p_value < alpha:
    print("Reject the null hypothesis: Home teams score significantly more goals than away teams.")
else:
    print("Fail to reject the null hypothesis: No significant difference in goals scored by home and away teams.")

T-Test Results for Home Advantage:
T-Statistic: 10.168129153700171
P-Value: 3.6926276633557134e-24
Reject the null hypothesis: Home teams score significantly more goals than away teams.


# INTERPRETATION : The T-test results show that home teams score significantly more goals than away teams. The T-statistic of 10.17 and a very small P-value (3.69e-24) lead to the rejection of the null hypothesis, indicating a strong home advantage in goal scoring.